cheetah/accelerator.py

```diff
@@ -299,6 +307,354 @@ def defining_features(self) -> list[str]:
     def __repr__(self) -> str:
         return f"{self.__class__.__name__}(length={repr(self.length)})"
 
+class SpaceChargeKick(Element):
+    """
+    Simulates space charge effects on a beam.
+    :param length: Length of the element in meters.
+    :param grid_precision: Number of grid points in each dimension.
+    :param grid_dimensions: Dimensions of the grid as multiples of sigma of the beam.
```

Comment:
Maybe `num_grid_points_x`, `num_grid_points_y`, `num_grid_points_z` ... and `grid_extend_x`, `grid_extend_y`, `grid_extend_z`. I generally think names should be chose such that someone unfamiliar with the code doesn't always need to scroll back to the docstring to read and understand what variables are.

Reply:
Maybe it would also be easier to have a 3D tensor for each and then just call them `num_grid_points` and `grid_extend`?

I think both ways have their pros and cons. With `Quadrupole.misalignment` or `Screen.resolution` we currently use the one Tensor approach a lot in Cheetah. (One day it might be nice to also change of these to dataclasses, so you can do `num_grid_points.x` ... but that's for us to solve in the future, not now.)